src/EditorFeatures/Core.Cocoa/Preview/PreviewFactoryService.cs

```diff
@@ -51,11 +51,8 @@ protected override async Task<ICocoaDifferenceViewer> CreateDifferenceViewAsync(
             var diffViewer = _differenceViewerService.CreateDifferenceView(diffBuffer, previewRoleSet);
             diffViewer.ViewMode = mode;
 
-            // This code path must be invoked on UI thread.
-            AssertIsForeground();
```

Comment:
All the foreground assertions are now gone?

Reply:
Foreground assertions are unnecessary in asynchronous code. If code needs a specific thread, and can and should switch to that thread. The caller's thread is generally irrelevant.

Reply:
Yup, I saw some codepaths did not change to switch, but I assume this was because they're only called inside the UI thread context.

LGTM